src/main/java/tools/jackson/databind/deser/jdk/CollectionDeserializer.java

```diff
@@ -595,4 +598,18 @@ public void handleResolvedForwardReference(DeserializationContext ctxt, Object i
             _parent.resolveForwardReference(ctxt, id, value);
         }
     }
+
+    static Boolean findFormatFeatureOrClassFallback(DeserializationContext ctxt,
+        JavaType type, Boolean unwrapSingle, JsonFormat.Feature feature)
+    {
+        if (unwrapSingle != null) {
+            return unwrapSingle;
+        }
+        AnnotatedClass ac = ctxt.introspectClassAnnotations(type);
```

Comment:
Ideally we would not need to re-introspect (adds to startup overhead)... since `BasicDeserializerFactory.createCollectionDeserializer(...)` has `BeanDescription.Supplier beanDescRef` and could pass it to `CollectionDeserializer`.

That'd require a bit of refactoring that might be worth it: instead of direct call to constructor, should call factory method... I'll go ahead and do that part of refactoring now.

Reply:
Ok, #5525 added 

```
    public static CollectionDeserializer create(JavaType collectionType,
            BeanDescription.Supplier beanDescRef,
            ValueDeserializer<Object> valueDeser,
            TypeDeserializer valueTypeDeser, ValueInstantiator valueInstantiator)

```

where

    beanDescRef.getClassAnnotations();

can be used to get introspected class annotations of target type. This could be passed to constructor (new overload with added parameter), retained in a `transient` field, to be available for `createContextual()`. May sound like lots of work but annotation introspection is becoming bigger and bigger problem with start up overhead with Jackson so I'd like to limit additional calls to re-introspecting again and again.